### ISSUE TITLE
Cannot debug vue-hackernews-2.0

### ISSUE THREAD

**Issue by octref** [Microsoft/vscode-chrome-debug#430]
- VS Code Version: 1.12.1

I'm not quite sure which part went wrong, but it feels really close to working.

Steps to reproduce:

1. Clone https://github.com/vuejs/vue-hackernews-2.0 -> `yarn` -> `npm run dev`

2. Add debug option like so
    ```json
    {
      "version": "0.2.0",
      "configurations": [
        {
          "type": "chrome",
          "request": "launch",
          "name": "Launch Chrome against localhost",
          "url": "http://localhost:8080",
          "webRoot": "${workspaceRoot}",
          "sourceMaps": true
        }
      ]
    }
    ```

3. In chrome I'm able to debug it: 
    ![image](https://cloud.githubusercontent.com/assets/4033249/26028398/86ce9c8c-37ed-11e7-8b5f-e8ab04e84715.png)

4. In VSCode, some part are unable to set breakpoints on:

    ![image](https://cloud.githubusercontent.com/assets/4033249/26028413/c6e59b54-37ed-11e7-8502-ac9d614d93ef.png)

    Some other parts I can set breakpoints on, but VSCode doesn't stop on the breakpoints:

    ![image](https://cloud.githubusercontent.com/assets/4033249/26028423/fcd9d2ac-37ed-11e7-8dbd-bbcb3d9ecf50.png)

### Note:

- 3 & 4 are talking about line 58 and 82 for this file: https://github.com/vuejs/vue-hackernews-2.0/blob/master/src/views/ItemList.vue
- http://localhost:8080/json returns 404 as router is not setup with this path
- This is using `webpack-dev-server` with `devtool` as `#source-map`: https://github.com/vuejs/vue-hackernews-2.0/blob/master/build/webpack.server.config.js#L9

---

`.scripts`

```
› VM35
› http://localhost:8080/
› http://localhost:8080/dist/1.19a266defc74c06e9d7c.hot-update.js
    - webpack:///./src/views/ItemList.vue?d81b* (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///ItemList.vue?4794 (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
› http://localhost:8080/dist/1.bbbd4a966dec73dde7f3.hot-update.js
    - webpack:///ItemList.vue?8425 (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
› http://localhost:8080/dist/1.d6c90d103d32ce77919f.hot-update.js
    - webpack:///./src/views/ItemList.vue?d81b* (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///ItemList.vue?88ab (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
› http://localhost:8080/dist/1.js
    - webpack:///src/views/CreateListView.js (/Users/pine/Code/repo/vue-hackernews-2.0/views/CreateListView.js)
    - webpack:///./src/views/ItemList.vue?e1b5 (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///./src/views/ItemList.vue?d81b (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///./src/views/ItemList.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///./src/views/ItemList.vue?f9de (/Users/pine/Code/repo/vue-hackernews-2.0/src/views/ItemList.vue)
    - webpack:///./src/components/Item.vue?bf44 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue?6da0 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue?6624 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/Item.vue)
    - webpack:///ItemList.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
› http://localhost:8080/dist/app.js
    - webpack:///./src/components/ProgressBar.vue?2571 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/ProgressBar.vue)
    - webpack:///./src/components/ProgressBar.vue?cbce (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/ProgressBar.vue)
    - webpack:///./src/components/ProgressBar.vue?1685 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/ProgressBar.vue)
    - webpack:///./src/components/ProgressBar.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/ProgressBar.vue)
    - webpack:///./src/App.vue?df14 (/Users/pine/Code/repo/vue-hackernews-2.0/src/App.vue)
    - webpack:///./src/App.vue?b841 (/Users/pine/Code/repo/vue-hackernews-2.0/src/App.vue)
    - webpack:///./src/App.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/App.vue)
    - webpack:///./src/App.vue?1890 (/Users/pine/Code/repo/vue-hackernews-2.0/src/App.vue)
    - webpack:///src/entry-client.js (/Users/pine/Code/repo/vue-hackernews-2.0/entry-client.js)
    - webpack:///src/util/title.js (/Users/pine/Code/repo/vue-hackernews-2.0/util/title.js)
    - webpack:///src/api/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/api/index.js)
    - webpack:///src/util/filters.js (/Users/pine/Code/repo/vue-hackernews-2.0/util/filters.js)
    - webpack:///ProgressBar.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ProgressBar.vue)
    - webpack:///src/api/create-api-client.js (/Users/pine/Code/repo/vue-hackernews-2.0/api/create-api-client.js)
    - webpack:///src/app.js (/Users/pine/Code/repo/vue-hackernews-2.0/app.js)
    - webpack:///src/router/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/router/index.js)
    - webpack:///src/store/actions.js (/Users/pine/Code/repo/vue-hackernews-2.0/store/actions.js)
    - webpack:///src/store/getters.js (/Users/pine/Code/repo/vue-hackernews-2.0/store/getters.js)
    - webpack:///src/store/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/store/index.js)
    - webpack:///src/store/mutations.js (/Users/pine/Code/repo/vue-hackernews-2.0/store/mutations.js)
    - webpack:///./public/logo-48.png (/Users/pine/Code/repo/vue-hackernews-2.0/public/logo-48.png)
› http://localhost:8080/dist/manifest.js
    - webpack:///webpack/bootstrap d6c90d103d32ce77919f (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/webpack/bootstrap d6c90d103d32ce77919f)
› http://localhost:8080/dist/vendor.js
    - webpack:///./~/vue/dist/vue.runtime.esm.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue/dist/vue.runtime.esm.js)
    - webpack:///./~/vue-hot-reload-api/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue-hot-reload-api/index.js)
    - webpack:///(webpack)/buildin/global.js (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/(webpack)/buildin/global.js)
    - webpack:///./~/css-loader/lib/css-base.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/css-loader/lib/css-base.js)
    - webpack:///./~/vue-loader/lib/component-normalizer.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue-loader/lib/component-normalizer.js)
    - webpack:///./~/vue-style-loader/lib/addStylesClient.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue-style-loader/lib/addStylesClient.js)
    - webpack:///./~/firebase/app.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/firebase/app.js)
    - webpack:///./~/html-entities/lib/html5-entities.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/html-entities/lib/html5-entities.js)
    - webpack:///./~/process/browser.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/process/browser.js)
    - webpack:///./~/timers-browserify/main.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/timers-browserify/main.js)
    - webpack:///(webpack)-hot-middleware/client.js (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/(webpack)-hot-middleware/client.js)
    - webpack:///./~/ansi-html/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/ansi-html/index.js)
    - webpack:///./~/ansi-regex/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/ansi-regex/index.js)
    - webpack:///./~/base64-js/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/base64-js/index.js)
    - webpack:///./~/buffer/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/buffer/index.js)
    - webpack:///./~/es6-promise/auto.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/es6-promise/auto.js)
    - webpack:///./~/es6-promise/dist/es6-promise.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/es6-promise/dist/es6-promise.js)
    - webpack:///./~/firebase/database.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/firebase/database.js)
    - webpack:///./~/html-entities/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/html-entities/index.js)
    - webpack:///./~/html-entities/lib/html4-entities.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/html-entities/lib/html4-entities.js)
    - webpack:///./~/html-entities/lib/xml-entities.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/html-entities/lib/xml-entities.js)
    - webpack:///./~/ieee754/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/ieee754/index.js)
    - webpack:///./~/isarray/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/isarray/index.js)
    - webpack:///./~/setimmediate/setImmediate.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/setimmediate/setImmediate.js)
    - webpack:///./~/strip-ansi/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/strip-ansi/index.js)
    - webpack:///./~/vue-router/dist/vue-router.esm.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue-router/dist/vue-router.esm.js)
    - webpack:///./~/vue-style-loader/lib/listToStyles.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vue-style-loader/lib/listToStyles.js)
    - webpack:///./~/vuex-router-sync/index.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vuex-router-sync/index.js)
    - webpack:///./~/vuex/dist/vuex.esm.js (/Users/pine/Code/repo/vue-hackernews-2.0/node_modules/vuex/dist/vuex.esm.js)
    - webpack:///(webpack)-hot-middleware/client-overlay.js (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/(webpack)-hot-middleware/client-overlay.js)
    - webpack:///(webpack)-hot-middleware/process-update.js (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/(webpack)-hot-middleware/process-update.js)
    - webpack:///(webpack)/buildin/module.js (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/(webpack)/buildin/module.js)
```

**Comment by auchenberg** [comment]
Looking at your `.scripts` it looks like your source files are placed in `/src` right? 

Try to update your `"webRoot": "${workspaceRoot}/src"`

**Comment by octref** [comment]
I tried that before, didn't work either. Both says "Breakpoints ignored because generated code not found (source map problem?)".

![image](https://cloud.githubusercontent.com/assets/4033249/26028599/0c36da8e-37f2-11e7-9af9-046e9c48dbbf.png)

Because of output like that:

```
webpack:///./src/components/Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
```

So I thought it's a direct mapping, as both side has `src` in it and I wouldn't need it in `webRoot`.

**Comment by auchenberg** [comment]
For `ItemList.vue` the problem seems to be this mapping:

```
webpack:///ItemList.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
```

The problematic part: `/dist/webpack:/ItemList.vue`

Looks like you have to setup some sourceMapOverrides. See https://github.com/Microsoft/vscode-chrome-debug#sourcemaps (On my phone right now, so I can't debug it)

**Comment by octref** [comment]
I was trying that too, but for this one:

```
    - webpack:///Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/Item.vue)
    - webpack:///ItemList.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
```

`Item.vue` is in `src/components` whereas `ItemList.vue` is in `src/views`, so I'm not sure how can I set it up so both point to the right file.

**Comment by octref** [comment]
> On my phone right now, so I can't debug it

It's nothing critical. I can wait until later. Thanks for the help!

**Comment by roblourens** [comment]
Yeah you will need sourceMapPathOverrides. What did you try?

**Comment by octref** [comment]
```
 "webpack:///*": "${workspaceRoot}/src/components/*"
```

as in https://github.com/octref/vetur/issues/116#issuecomment-301273278, but it didn't work.

Also tried
```
 "webpack:///*": "${workspaceRoot}/src/views/*"
```

**Comment by octref** [comment]
Oops, somehow now it's somewhat working.

When I have `"webpack:///*": "${workspaceRoot}/src/components/*"` I can only debug items under `components`, when I have ` "webpack:///*": "${workspaceRoot}/src/views/*"` I can only debug those under `views`.

As my report above, these two lines:

```
- webpack:///Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/Item.vue)
- webpack:///ItemList.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/ItemList.vue)
```

`Item.vue` is in `components`, and `ItemList.vue` is in `views`. Can't seem to debug them together...

**Comment by roblourens** [comment]
It looks like there are two entries for Item.vue, like `webpack:///./src/components/Item.vue` and `webpack:///Item.vue`? Why is that? And why would this webpack setup write files from different folders to `webpack:///` - what if you have Item.vue in `components` and also in `views`?

**Comment by octref** [comment]
Yeah I don't understand this either:

```
    - webpack:///./src/components/Item.vue?bf44 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue?6da0 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue?6624 (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///./src/components/Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/src/components/Item.vue)
    - webpack:///Item.vue (/Users/pine/Code/repo/vue-hackernews-2.0/dist/webpack:/Item.vue)
```
The first few might be because of caching/hot-reloading, but the last entry doesn't make much sense...

**Comment by roblourens** [comment]
If you look it up in chrome devtools, is one of those the 'real' one? And I'm curious what happens if you rename ItemList to Item and see how it handles the collision.

**Comment by octref** [comment]
@roblourens You can run this pretty easily...https://github.com/vuejs/vue-hackernews-2.0 Just `npm install` and `npm run dev`.

As for your question, `webpack:///Item.vue` is the real one:

![image](https://cloud.githubusercontent.com/assets/4033249/26037269/d1879fb4-38bd-11e7-8f46-87e745dda4c4.png)

`webpack:///./src/components/Item.vue`:

![image](https://cloud.githubusercontent.com/assets/4033249/26037271/ddb8f2d8-38bd-11e7-8985-4bf01336526e.png)

**Comment by auchenberg** [comment]
Looks like VueJS + WebPack generates separate "chunks" for the JS, HTML and CSS, which confuses the sourcemap.

**Comment by octref** [comment]
Yea the config is pretty crazy: https://github.com/vuejs/vue-hackernews-2.0/blob/master/build/webpack.client.config.js

Originally I thought Chrome debugs it fine but it's chocking on some parts, too.

I'll try simplify the webpack config to make it debuggable. Meanwhile, @auchenberg and @roblourens, in your experience, which of the webpack [devtool option](https://webpack.js.org/configuration/devtool/) generates sourcemap that works best with VSCode + Chrome Debug?

**Comment by auchenberg** [comment]
@octref We usually recommend "source-map" or "cheap-source-map". We work out of the box with react-boilerplate and create-react-app for example: https://github.com/react-boilerplate/react-boilerplate

**Comment by roblourens** [comment]
I can sort of set breakpoints if I use 'source-map' and

```
"sourceMapPathOverrides": {
    "webpack:///*": "${workspaceRoot}/src/views/*"
}
```

which is just a workaround for ItemList. Still some flakiness. I think the sourcemap name mangling is from the hot module reloader. Really crazy, I don't even know how to come up with a good general solution for this.

**Comment by layanto** [comment]
I have exactly the same issue where vue components located in different directories (src/components/ and src/views) but appear as if they are in the same directory in webpack:///

**Comment by layanto** [comment]
To debug in chrome devtool (not vscode), I use the one without the full path. The one without the full path shows the source of the vue single file component while the one with the full path appears to display the compiled javascript.

**Comment by layanto** [comment]
I don't think this can be solved by vscode-chrome-debug as there is no way sourceMapPathOverrides can correctly find the location of the source file when two sources located in two different directories have the same webpack:/// path pattern. Issue must be with vue-loader and/or webpack.

**Comment by CXHtml** [comment]
It seems that we should first override the source map path explicitly in order to distinguish the cases mentioned above ( two different directories point to the same webpack:/// path pattern ) and then consider the common path pattern like the following in launch.json:

```
"sourceMapPathOverrides": {
     "webpack:///src/app.js": "${webRoot}/src/App.js",
     "webpack:///App.vue": "${webRoot}/src/App.vue",
     "webpack:///ItemList.vue": "${webRoot}/src/views/ItemList.vue",
     "webpack:///ItemView.vue": "${webRoot}/src/views/ItemView.vue",
     "webpack:///UserView.vue": "${webRoot}/src/views/UserView.vue",
     "webpack:///*.vue": "${webRoot}/src/components/*.vue",
     "webpack:///*": "${webRoot}/*"
}
```
This does work for vue-hackernews-2.0 examples @octref mentioned before, the issue is related to vue-loader or webpack?

**Comment by roblourens** [comment]
That's a pretty annoying workaround but I'm glad it works. I think it's from vue-loader, since I haven't seen this in any other webpack project.

**Comment by auchenberg** [comment]
Blocked on Webpack and vue-loader. We need a correct sourcemap before we can do anything.

**Comment by blaluc** [comment]
Any progress on this issue?

**Comment by auchenberg** [comment]
This issue in vue-land still needs be resolved, https://github.com/vuejs/vue-loader/issues/620,  before our debugger will work

**Comment by StephenSeraph** [comment]
Today I have some practice about this, and it finally works to me .
May it is useful to you.
[vue-debug-by-vscode](https://github.com/StephenSeraph/vue-debug-by-vscode)

**Comment by auchenberg** [comment]
You can now debug [`vue-cli`](https://github.com/vuejs/vue-cli) apps using the following `launch.json`:

```
{
    "type": "chrome",
    "request": "launch",
    "name": "Launch Chrome against localhost",
    "url": "http://localhost:8080",
    "webRoot": "${workspaceFolder}/src",
    "sourceMapPathOverrides": {
        "webpack:///src/*": "${webRoot}/*",
    }
}
```

And by updating `devtool: 'source-map'` in `dev` within `config/index.js` to update the webpack config.

**Comment by CXHtml** [comment]
Thanks to @auchenberg for giving me a hint to finally discover a way to debug not only just *.vue file in vscode, but also the *.js in the project (code such as vuex actions/getters/mutaitions and any other imported js files and even the code in node_modules need to debug as well).

First, based on @auchenberg's suggestion, we need to(highly recommended!) add path overrides for the code in node modules as well: 

```
{
    "type": "chrome",
    "request": "launch",
    "name": "Launch Chrome against localhost",
    "url": "http://localhost:8080",
    "webRoot": "${workspaceFolder}",
    "sourceMapPathOverrides": {
        "webpack:///src/*": "${webRoot}/src/*",
        "webpack:///./node_modules/*": "${webRoot}/node_modules/*"
    }
}
```
And come back to webpack config, I suggest use `devtool: 'eval-source-map'` which is suited for development, and in order to debug other *.js files in vue-hackernews-2.0, for example, you can change the 'real' sourcemap path of those js files to path `webpack:///src/*` by adding `devtoolModuleFilenameTemplate` in `output` like this:

```
output: {
    path: path.resolve(__dirname, '../dist'),
    publicPath: '/dist/',
    filename: '[name].[chunkhash].js',
    devtoolModuleFilenameTemplate: info => {
      const isJsFile = info.resourcePath.match(/^\.\/src\/.*\.js$/)
      return `webpack:///${isJsFile ? info.resourcePath.replace('./', '') : info.resourcePath}`
    }
  },
```
now, we can debug all the code that run in browser under the correct sourcemap path `webpack:///src/*`:

![image](https://user-images.githubusercontent.com/6263570/34719893-9a5554e2-f577-11e7-9a68-c20d1833f066.png)

type `.scripts`  in vscode debug console, you should get correct map for files above :)

**Comment by roblourens** [comment]
Are you sure you need that `devtoolModuleFilenameTemplate`? Could you just handle that in `sourceMapPathOverrides`, like `"webpack:///./src/*": "${webRoot}/src/*"`?

**Comment by CXHtml** [comment]
If I add `webpack:///./src/*": "${webRoot}/src/*` and `webpack:///src/*": "${webRoot}/src/*` at the same time we will get this problem, for example:

```
› webpack-internal:///168
    - webpack:///./src/components/Item.vue?08ce (/Users/cxhtml/vue-hackernews-2.0/src/components/Item.vue)
› webpack-internal:///169
    - webpack:///./src/components/Item.vue?85ec (/Users/cxhtml/vue-hackernews-2.0/src/components/Item.vue)
› webpack-internal:///170
    - webpack:///src/components/Item.vue?da82 (/Users/cxhtml/vue-hackernews-2.0/src/components/Item.vue)
› webpack-internal:///171
    - webpack:///./src/components/Item.vue?c9a8 (/Users/cxhtml/vue-hackernews-2.0/src/components/Item.vue)
```
you can see multiple `Item.vue` files map to the same path, but only the `webpack:///src/components/Item.vue?da82 ` is as same as the file I edit in vscode, so in this case, If I set a breakpoint for Item.vue, it may not works. Therefore, in my suggestion, I just consider one case that all the debuggable code are placed in `webpack:///src/*` and `devtoolModuleFilenameTemplate` can help me solve this problem

I think the reason why multiple same sourcemap files generated from webpack is that in vue loader, it extracts html/js/style from *.vue and apply different loaders respectively and sourcemap is also generated during each loader processing.

**Comment by roblourens** [comment]
What you say is right, but I thought the `devtoolModuleFilenameTemplate` is remapping the `webpack:///./src` files to `webpack:///src` or something, and since there are already files with the same sourcemap path, I don't really know what that would do.

**Comment by CXHtml** [comment]
My `devtoolModuleFilenameTemplate` is just remapping the *.js files from `webpack:///./src` to `webpack:///src` and since **only** the *.vue files have 'the same sourcemap path' case, I only map `webpack:///src/**/*.vue` to the absolute path of the corresponding file, in other word, all the code I want to debug are now in `webpack:///src/*`, and there will be no more files with the same sourcemap path exist. 

I know using `devtoolModuleFilenameTemplate ` may looks weird, but I cannot find other way unless you tell me this glob pattern can work in `sourceMapPathOverrides`:

```
"sourceMapPathOverrides": {
                "webpack:///./src/**/*.js": "${webRoot}/src/**/*.js"
}
```
instead of using `devtoolModuleFilenameTemplate`.

**Comment by roblourens** [comment]
Can you use 

```
"sourceMapPathOverrides": {
                "webpack:///./src/*": "${webRoot}/src/*"
}
```
?

**Comment by octref** [comment]
@auchenberg Thanks a lot. This works fine for me during testing, other than an off-by-one line betweeen Chrome / VS Code.

Looking forward to your guide. But I'm wondering if i should include some instructions in Vetur doc too.

![vue-debug](https://user-images.githubusercontent.com/4033249/35004201-0989a058-faa4-11e7-93a2-fc4c77e7ab16.gif)

**Comment by liuliangsir** [comment]
Hi, guys, the reason that failure to debug vue file in vscode  is this evil mapping. so for the problematic part, you could set up some `sourceMapPathOverrides ` .  
The following code snippets are used for vscode and webpack, as it follows:

```
// launch.json
{
    // 使用 IntelliSense 了解相关属性。
    // 悬停以查看现有属性的描述。
    // 欲了解更多信息，请访问: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
          "name": "调试浏览器端代码",
          "type": "chrome",
          "request": "launch",
          "webRoot": "${workspaceRoot}",
          "url": "http://static.yizhibo.com/html/question/index.html",
          // "url": "http://localhost:8080",
          "userDataDir": "${workspaceRoot}/.vscode/chrome",
          "sourceMaps": true,
          "preLaunchTask": null,
          "smartStep": true,
          "internalConsoleOptions": "openOnSessionStart",
          "skipFiles": [
            "node_modules/**"
          ],
          "sourceMapPathOverrides": {
            // "webpack:///src/app.js": "${webRoot}/src/App.js",
            "webpack:///App.vue": "${webRoot}/src/App.vue",
            // "webpack:///ItemList.vue": "${webRoot}/src/views/ItemList.vue",
            // "webpack:///ItemView.vue": "${webRoot}/src/views/ItemView.vue",
            // "webpack:///UserView.vue": "${webRoot}/src/views/UserView.vue",
            "webpack:///*.vue": "${webRoot}/src/components/*.vue",
            "webpack:///*": "${webRoot}/*"
          }
        }
    ]
}

```
---------------------------------------------split line-----------------------------------------------

```
//webpack config 
{
  output: {
    path: config.build.assetsRoot,
    filename: '[name].js',
    publicPath: process.env.NODE_ENV === 'production'
      ? config.build.assetsPublicPath
      : config.dev.assetsPublicPath,
    devtoolModuleFilenameTemplate(info) {
      return 'file:///' + encodeURI(info.absoluteResourcePath)
    },
    devtoolFallbackModuleFilenameTemplate(info) {
      return 'file:///' + encodeURI(info.absoluteResourcePath) + '?' + info.hash
    }
  }
}
```
BTW, i use vue-cli to init project

**Comment by auchenberg** [comment]
Closing this as we have landed Vue.js support via vue-cli, and I've written a debugging recipe for this in https://github.com/Microsoft/vscode-recipes/tree/master/vuejs-cli. 

I can't get the `vue-hackernews-2.0` working in neither VS Code or Chrome Devtools, as there seem to be a deeper problem with sourcemaps when using server-side-rendering. That needs to get fixed before we can do anything here on VS Code.
